include/seqan3/search/kmer_index/shape.hpp

```diff
@@ -0,0 +1,141 @@
+// -----------------------------------------------------------------------------------------------------
+// Copyright (c) 2006-2019, Knut Reinert & Freie Universität Berlin
+// Copyright (c) 2016-2019, Knut Reinert & MPI für molekulare Genetik
+// This file may be used, modified and/or redistributed under the terms of the 3-clause BSD-License
+// shipped with this file and also available at: https://github.com/seqan/seqan3/blob/master/LICENSE.md
+// -----------------------------------------------------------------------------------------------------
+
+/*!\file
+ * \author Enrico Seiler <enrico.seiler AT fu-berlin.de>
+ * \author Vinzenz May <vinzenz.may AT fu-berlin.de>
+ * \brief Provides seqan3::shape.
+ */
+
+#pragma once
+
+#include <seqan3/core/detail/strong_type.hpp>
+#include <seqan3/range/container/small_vector.hpp>
+
+namespace seqan3
+{
+    
+/*!\brief A strong type of underlying type `uint8_t` that represents the ungapped shape size.
+ * \tparam value_t The size of the shape
+ */
+struct ungapped : detail::strong_type<uint8_t, ungapped>
+{
+    using detail::strong_type<uint8_t, ungapped>::strong_type;
+    
+    //!\brief Construct from an unsigned integral.
+    template<std::UnsignedIntegral t>
+    //!\cond
+        requires !std::Same<t, uint8_t>
+    //!\endcond
+    constexpr ungapped(t value) noexcept: ungapped(static_cast<uint8_t>(value))
+    {
+        assert(value <= 255);
```

Comment:
But actually only 30 is allowed, right?

Reply:
Generally yes, but I think the strong_type should just check if the input is valid